src/bin/main.rs

```diff
@@ -0,0 +1,127 @@
+extern crate observer;
+use chrono::Utc;
+use core::borrow::{Borrow, BorrowMut};
+use observer::context::{Context, Frame};
+use observer::event::Event;
+use observer::observer::{observe, save};
+use observer::AResult;
+use serde::ser::Serialize;
+use serde_derive::{Deserialize, Serialize};
+use std::collections::LinkedList;
+use std::thread;
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct UserId {
+    id: i32,
+}
+
+impl Event<UserId> for UserId {
+    fn map(&self, ctx: &RefCell<Context>, _data: UserId) -> AResult<UserId> {
+        Ok(_data)
+    }
+
+    fn name(&self) -> String {
+        "UserId".to_string()
+    }
+
+    fn destination(&self) -> String {
+        "user_id".to_string()
+    }
+}
```

Comment:
What are name and destination functions for?

Reply:
destination function is for getting folder name on local file system to save data of observe for particular structs. used name function to create frame_id.